src/MiniProfiler.Shared/ui/lib/MiniProfiler.ts

```diff
@@ -886,125 +899,302 @@ namespace StackExchange.Profiling {
+                                popup.style.maxHeight = 'calc(100vh - ' + (bottom + 25) + 'px)';
                             } else {
-                                popup.css({ 'top': 0, 'max-height': 'calc(100vh - ' + (button.offset().top - $(window).scrollTop() + 25) + 'px)' });
+                                popup.style.top = '0';
+                                popup.style.maxHeight = 'calc(100vh - ' + (button.getBoundingClientRect().top - window.window.scrollY + 25) + 'px)';
                             }
                         }
-                    }).on('click', '.mp-queries-show', function(e) {
+                        return;
+                    }
+                    const queriesButton = target.closest('.mp-queries-show') as HTMLElement;
+                    if (queriesButton) {
                         // opaque background
-                        const overlay = $('<div class="mp-overlay"><div class="mp-overlay-bg"/></div>').appendTo('body');
-                        const queries = $(this).closest('.mp-result').find('.mp-queries').clone().appendTo(overlay).show();
+                        document.body.insertAdjacentHTML('beforeend', '<div class="mp-overlay"><div class="mp-overlay-bg"/></div>');
+                        const overlay = document.querySelector('.mp-overlay');
+                        const queriesOrig = queriesButton.closest('.mp-result').querySelector('.mp-queries');
+                        const queries = queriesOrig.cloneNode(true) as HTMLDivElement;
+                        queries.style.display = 'block';
+                        overlay.appendChild(queries);
 
-                        mp.scrollToQuery($(this), queries, queries);
+                        mp.scrollToQuery(queriesButton, queries, queries);
 
                         // syntax highlighting
-                        queries.find('pre code').each((i, block) => mp.highlight(block));
-                    }).on('click keyup', (e) => {
-                        const active = $('.mp-result.active');
-                        if (active.length) {
-                            const bg = $('.mp-overlay');
-                            const isEscPress = e.type === 'keyup' && e.which === 27;
-                            const isBgClick = e.type === 'click' && !$(e.target).closest('.mp-queries, .mp-results').length;
-
-                            if (isEscPress || isBgClick) {
-                                if (bg.is(':visible')) {
-                                    bg.remove();
-                                } else {
-                                    active.removeClass('active');
+                        queries.querySelectorAll('pre code').forEach(block => mp.highlight(block as HTMLElement));
+                        return;
+                    }
+                });
+                // Background and esc binding to close popups
+                const tryCloseActive = (event: MouseEvent | KeyboardEvent) => {
```

Comment:
Other code (including new code in this PR) uses `function`; this uses `() => {}`. Probably best to standardise on one.

Does this get transpiled to ES5 before it gets to browsers? If not, ES6 shorthand syntax loses you IE support.

You may know all of this already, of course :)

Reply:
A fair point! On this one, it should be `function tryCloseActive()` for consistency.

On why I'm not using a specific one: arrow functions don't bind `this` in the same way and we want to alter the element in the handler, e.g. `this.classList` - so I standardize on `function X()` for event handlers, and arrow functions everywhere else. Making that adjustment here :)

Reply:
Took a look - normalized the others but we're in a block here and we can't declare it function style (and don't need `this`), so leaving as is for this case